### PR TITLE
add sphinx-lint as pre-commit

### LINKED ISSUE #1640: Problem: sphinx-lint returns linting errors in the doc
**Version of the documentation**

1.15

**Page (and section, if applicable) where the issue occurs**

Admin manual and Getting started sections, and in some localization projects.

**Description of the issue**

Using `sphinx-lint` - https://github.com/sphinx-contrib/sphinx-lint#using-sphinx-lint - on the AM docs repo helped me find a few linting issues.

Some of them I am able to fix - see the PR - but others are in locales and I am not sure if we should address those in the docs repo directly or in Weblate.

**Suggested fix**

See the PR for those I was able to fix.

Maybe we could add sphinx-lint to the pre-commit for future use ?

---

**For Artefactual use:**

Before you close this issue, you must check off the following:

- [ ] All pull requests related to this issue are properly linked
- [ ] All pull requests related to this issue have been merged
- [ ] Details about this issue have been added to the release notes (if applicable)

### PR DESCRIPTION
**Pull request by eviau-artefactual**
Closes https://github.com/archivematica/Issues/issues/1640

### CODE REVIEW

**Review by replaceafill — COMMENTED** [review]
@eviau-artefactual I ran `pre-commit` on your branch and got several linting errors:

```console
(venv310) replaceafill@vm1:~/archivematica-docs$ pre-commit run --all-files
Sphinx Lint..............................................................Failed
- hook id: sphinx-lint
- exit code: 1

admin-manual/installation-setup/integrations/atom-setup.rst:42: Line ends with dangling hyphen (dangling-hyphen)
admin-manual/security/security.rst:257: found an unbalanced inline literal markup. (unbalanced-inline-literals-delimiters)
admin-manual/security/security.rst:29: Line ends with dangling hyphen (dangling-hyphen)
dev-manual/api/api-reference-archivematica.rst:61: Line ends with dangling hyphen (dangling-hyphen)
dev-manual/api/api-reference-archivematica.rst:192: Line ends with dangling hyphen (dangling-hyphen)
dev-manual/api/api-reference-archivematica.rst:193: Line ends with dangling hyphen (dangling-hyphen)
dev-manual/api/api-reference-archivematica.rst:203: Line ends with dangling hyphen (dangling-hyphen)
dev-manual/api/api-reference-archivematica.rst:336: Line ends with dangling hyphen (dangling-hyphen)
dev-manual/api/api-reference-storage-service.rst:810: Line ends with dangling hyphen (dangling-hyphen)
getting-started/troubleshooting/faq.rst:36: Line ends with dangling hyphen (dangling-hyphen)
getting-started/troubleshooting/error-handling.rst:272: Line ends with dangling hyphen (dangling-hyphen)
user-manual/access/access.rst:94: trailing whitespace (trailing-whitespace)
user-manual/access/access.rst:129: Line ends with dangling hyphen (dangling-hyphen)
user-manual/administer/dashboard-admin.rst:412: Line ends with dangling hyphen (dangling-hyphen)
user-manual/archival-storage/archival-storage.rst:11: trailing whitespace (trailing-whitespace)
user-manual/archival-storage/archival-storage.rst:60: trailing whitespace (trailing-whitespace)
user-manual/archival-storage/archival-storage.rst:70: trailing whitespace (trailing-whitespace)
user-manual/archival-storage/archival-storage.rst:80: trailing whitespace (trailing-whitespace)
user-manual/archival-storage/archival-storage.rst:104: trailing whitespace (trailing-whitespace)
user-manual/archival-storage/archival-storage.rst:122: trailing whitespace (trailing-whitespace)
user-manual/archival-storage/archival-storage.rst:136: trailing whitespace (trailing-whitespace)
user-manual/archival-storage/archival-storage.rst:179: trailing whitespace (trailing-whitespace)
```

Do you plan to address those in this pull request?

Also, now that we have this hook do you think we should include it in [our CI test workflow](https://github.com/artefactual/archivematica-docs/blob/bf3b2936bcac93e8fd775e0fb3548ac6759ffd3d/.github/workflows/test.yml)? That'd guarantee that new linting problems are not introduced if people don't install or run `pre-commit` in their environments.

**Comment by eviau-artefactual** [comment]
We might want to make the switch to 'list-table' directives instead of full tables... because of https://github.com/sphinx-contrib/sphinx-lint#known-issues

We could also not use sphinx-lint, or (try to) configure it so it ignores tables.

**Comment by eviau-artefactual** [comment]
> Do you plan to address those in this pull request?

Last few commits should have solved everything. I did had to had the list-table directives, let me know if this solution works or not... ? thank you for catching the linting errors !

> Also, now that we have this hook do you think we should include it in [our CI test workflow](https://github.com/artefactual/archivematica-docs/blob/bf3b2936bcac93e8fd775e0fb3548ac6759ffd3d/.github/workflows/test.yml)? That'd guarantee that new linting problems are not introduced if people don't install or run `pre-commit` in their environments.

Yes, I think this would be a good idea. Should I open another PR or is it good enough to add these changes to this one?

**Comment by replaceafill** [comment]
> I did had to had the list-table directives, let me know if this solution works or not... ?

That looks fine to me.

> Should I open another PR or is it good enough to add these changes to this one?

I'd prefer adding the changes to this one.

**Comment by eviau-artefactual** [comment]
I added the pre-commit as an Action in `pre-commit.yml` - let me know if that works for you.

This complete this PR I believe - if anything is missing let me know.

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
```
Comment (replaceafill) [inline .github/workflows/pre-commit.yml]: This repository doesn't have a `main` branch.

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v3
```
Comment (replaceafill) [inline .github/workflows/pre-commit.yml]:
```suggestion
    - uses: actions/checkout@v4
    - uses: actions/setup-python@v5
```

**Review by replaceafill — COMMENTED** [review]
> I added the pre-commit as an Action in pre-commit.yml - let me know if that works for you.

In most of the Archivematica related repositories we have the lint job in the `test.yml` workflow:

* [archivematica](https://github.com/artefactual/archivematica/blob/43d46f3a735def6d888e77b34815028086ffda12/.github/workflows/test.yml#L94-L113)
* [amclient](https://github.com/artefactual-labs/amclient/blob/a2b33f20e8dd721ddb4ab8c7ea0b1a3cc46636e0/.github/workflows/test.yml#L47-L67)
* [mets-reader-writer](https://github.com/artefactual-labs/mets-reader-writer/blob/544312cbc4a8d8144d7c5b9fe3f6ff61fc34949d/.github/workflows/test.yml#L47-L67)

What do you think of keeping a single workflow with two jobs?

File: .github/workflows/pre-commit.yml (hunk 2 above)
Comment (eviau-artefactual) [reply .github/workflows/pre-commit.yml]: I added the change directly.

File: .github/workflows/pre-commit.yml (hunk 1 above)
Comment (eviau-artefactual) [reply .github/workflows/pre-commit.yml]: Oh, right.

**Comment by eviau-artefactual** [comment]
Ok, I think I got this to work! Let me know if something else is missing :)

File: .github/workflows/pre-commit.yml (hunk 2 above)
Comment (replaceafill) [reply .github/workflows/pre-commit.yml]: Could we use `actions/setup-python@v5`?

Comment (eviau-artefactual) [reply .github/workflows/pre-commit.yml]: Yes, I will make the change.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -32,3 +32,9 @@ jobs:
       - name: "Run build test"
         run: |
           make test
+  pre-commit:
```
Comment (replaceafill) [inline .github/workflows/test.yml]: In other AM related repositories we call the equivalent of this job `lint` or `linting` and we set a human readable `name` attribute for it and its steps. Could we do that here?

**Review by replaceafill — APPROVED** [review]
Thanks @eviau-artefactual!

**Pull request merged**